Sources/Moya/Endpoint.swift

```diff
@@ -99,7 +69,7 @@ extension Endpoint {
         request.httpMethod = method.rawValue
         request.allHTTPHeaderFields = httpHeaderFields
 
-        return try? parameterEncoding.encode(request, with: parameters)
+        return request
```

Comment:
This is a place where you would do a switch and prepare the request based on a task. We don't really want to do it in the core file since preparing the request based on an `Endpoint` is `Endpoint`'s responsibility. Another benefit is the fact that when we return nil there is an error thrown in basic request mapping and we don't even have to do a request ([source](https://github.com/Moya/Moya/blob/master/Sources/Moya/MoyaProvider%2BDefaults.swift#L16-L22)).